lbrynet/core/client/ConnectionManager.py

```diff
@@ -147,27 +152,27 @@ def protocol_disconnected(self, peer, protocol):
             del self._connections_closing[peer]
             d.callback(True)
 
+    def manage(self):
+        self._manage_deferred = self._manage()
+        if not self.stopped:
+            self._next_manage_call = self.callLater(self.MANAGE_CALL_INTERVAL_SEC, self.manage)
+
+
     @defer.inlineCallbacks
     def _manage(self):
-        self._manage_deferred = defer.Deferred()
-
-        from twisted.internet import reactor
         if len(self._peer_connections) < conf.settings['max_connections_per_stream']:
             log.debug("%s have %d connections, looking for %d",
                         self._get_log_name(), len(self._peer_connections),
                         conf.settings['max_connections_per_stream'])
+            ordered_request_creators = self._rank_request_creator_connections()
+            peers = yield self._get_new_peers(ordered_request_creators)
+            peer = self._pick_best_peer(peers)
             try:
-                ordered_request_creators = self._rank_request_creator_connections()
-                peers = yield self._get_new_peers(ordered_request_creators)
-                peer = self._pick_best_peer(peers)
                 yield self._connect_to_peer(peer)
             except Exception:
                 # log this otherwise it will just end up as an unhandled error in deferred
-                log.exception('Something bad happened picking a peer')
-        self._manage_deferred.callback(None)
-        self._manage_deferred = None
-        if not self.stopped:
-            self._next_manage_call = reactor.callLater(1, self._manage)
+                # Can happen if connection fails with ConnectionRefusedError
```

Comment:
yea I'll check all the exceptions that can be returned here , and see if it needs to be handled